Modify the blink(1) recipe to allow direct lookup of serial number

coledot opened the pull request:
Handy for assigning a specific blink(1) to a specific task.

coledot committed:
add the ability to identify a light by serial number

coledot committed:
revert version file

perryn commented:
Hi @coledot 

Thanks for the pull request!

A few questions/concerns
- I'm not sure how the code changes relate to the title of the pull request - all you seem to have done is move where the device is opened? Can you explain further?
- If you are going to modify tested code, please make sure you update the specs too and make sure they pass
- There are a number of changes included that just seem to be white space. This makes it difficult to read the pull request. Can you clean that up?

cheers
Perryn

coledot committed:
fix up specs & clean up whitespace

coledot commented:
Hi @perryn,

I modified the Light code to track both the handle and device for a given blink(1), instead of just the handle. Also, in the blink_1.rb recipe, I added a `#serial_number` method. This method is simply a wrapper around `@device.serial_number`. This allows one to say `light.serial_number` to get the blink(1) serial. `@handle` doesn't provide an equivalent method, from what I can see.

I've fixed the specs as well. The whitespace modifications are to remove excess spaces from the end of lines--just a minor cleanup/annoyance I wanted to deal with.
